MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```diff
@@ -20,15 +20,30 @@ final class FeedViewModel {
 	
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onErrorStateChange: Observer<String?>?
+	
 	
 	func loadFeed() {
+		self.onErrorStateChange?(nil)
 		onLoadingStateChange?(true)
 		feedLoader.load { [weak self] result in
-			if let feed = try? result.get() {
-				self?.onFeedLoad?(feed)
+			guard let self = self else { return }
+			switch result {
+			case .success(let feeds):
+				self.onFeedLoad?(feeds)
+			case .failure(_):
```

Comment:
`_` can be omitted here:

```suggestion
			case .failure:
```